src/model.cpp

```diff
@@ -130,23 +130,42 @@ Model::Model(const int nx_rdata, const int nxtrue_rdata, const int nx_solver,
        if Python wrapped: dxdotdp_explicit and dxdotdp_implicit are CSC matrices
      */
     if (pythonGenerated) {
-        dxdotdp_explicit =
-            SUNMatrixWrapper(nx_solver, static_cast<int>(p.size()),
-                             ndxdotdp_explicit, CSC_MAT);
+
+        dwdw_ = SUNMatrixWrapper(nw, nw, ndwdw, CSC_MAT);
+        // size dynamically adapted for dwdx_ and dwdp_
+        dwdx_ = SUNMatrixWrapper(nw, nx_solver, 0, CSC_MAT);
+        dwdp_ = SUNMatrixWrapper(nw, static_cast<int>(p.size()), 0, CSC_MAT);
+
+        for (int irec = 0; irec <= w_recursion_depth_; ++irec) {
+            dwdp_hierarchical_.emplace_back(
+                SUNMatrixWrapper(nw, static_cast<int>(p.size()), ndwdw + ndwdp,
```

Comment:
We can also add an additional constructor using the std::vector size type to get rid of all the static casts...